src/main/java/com/twilio/oai/resolver/common/CodegenModelContainerDataTypeResolver.java

```diff
@@ -0,0 +1,78 @@
+package com.twilio.oai.resolver.common;
+
+import com.twilio.oai.common.ApplicationConstants;
+import com.twilio.oai.common.LanguageDataType;
+import com.twilio.oai.resolver.Resolver;
+import org.openapitools.codegen.CodegenProperty;
+
+import java.util.List;
+import java.util.Stack;
+
+public class CodegenModelContainerDataTypeResolver implements Resolver<CodegenProperty> {
+
+    private CodegenModelDataTypeResolver codegenModelDataTypeResolver;
+    private List<? extends LanguageDataType> languageDataTypes;
+
+    public CodegenModelContainerDataTypeResolver(CodegenModelDataTypeResolver codegenModelDataTypeResolver,
+                                                 List<? extends LanguageDataType> languageDataTypes) {
+        this.codegenModelDataTypeResolver = codegenModelDataTypeResolver;
+        this.languageDataTypes = languageDataTypes;
+    }
+
+    public CodegenProperty resolve(CodegenProperty codegenProperty) {
+        Stack<String> containerTypes = new Stack<>();
+
+        codegenProperty.dataType = unwrapContainerType(codegenProperty,containerTypes);
```

Comment:
CodegenProperty has a [mostInnerItems property](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenProperty.java#L168) that may be able to replace the unwrapping and rewrapping of the data type. Something to consider

Reply:
Looked into this, but the `mostInnerItems` property `dataType` is different from the `CodegenProperty.dataType`. Not sure why.